pkg/build/api/validation/validation.go

```diff
@@ -96,6 +97,17 @@ func validateSource(input *buildapi.BuildSource) fielderrors.ValidationErrorList
 	return allErrs
 }
 
+func validateSecretRef(ref *kapi.LocalObjectReference) fielderrors.ValidationErrorList {
+	allErrs := fielderrors.ValidationErrorList{}
+	if ref == nil {
+		return allErrs
+	}
+	if len(ref.Name) == 0 {
+		allErrs = append(allErrs, fielderrors.NewFieldRequired("secret.name"))
```

Comment:
this should return `NewFieldRequired("name")` and the callers should prefix with the field name from the context